apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatView.kt

```diff
@@ -204,23 +467,75 @@ private fun InviteView(rhId: Long?, connReqInvitation: String, contactConnection
     SimpleXLinkQRCode(connReqInvitation, onShare = { chatModel.markShowingInvitationUsed() })
   }
 
-  Spacer(Modifier.height(10.dp))
-  val incognito = remember { mutableStateOf(controller.appPrefs.incognito.get()) }
-  IncognitoToggle(controller.appPrefs.incognito, incognito) {
-    ModalManager.start.showModal { IncognitoView() }
+  Spacer(Modifier.height(DEFAULT_PADDING))
+  val incognito by remember(chatModel.showingInvitation.value?.conn?.incognito, controller.appPrefs.incognito.get()) {
+    derivedStateOf {
+      chatModel.showingInvitation.value?.conn?.incognito ?: controller.appPrefs.incognito.get()
+    }
   }
-  KeyChangeEffect(incognito.value) {
-    withBGApi {
-      val contactConn = contactConnection.value ?: return@withBGApi
-      val conn = controller.apiSetConnectionIncognito(rhId, contactConn.pccConnId, incognito.value) ?: return@withBGApi
-      withChats {
-        contactConnection.value = conn
-        updateContactConnection(rhId, conn)
+
+  val currentUser by remember(chatModel.currentUser.value) {
+    derivedStateOf { chatModel.currentUser.value }
+  }
+
+  currentUser?.let {
```

Comment:
the block of code is quite large so if you want to use .let here, specify it like so at least:
```kotlin
currentUser?.let { user ->
```

it will be easier to track that you want to use the value, not just checking for non-null this way. 
Also we try to not use .let at all because it makes some people crazy when they read it:)
So this is better:
```kotlin
val user = currentUser.value
if (user != null) {
...
}
```

Reply:
```kotlin
val currentUser = remember { chatModel.currentUser }.value
  if (currentUser != null) {
```